src/cpu/cpu_batch_normalization_list.cpp

```diff
@@ -42,51 +42,54 @@ namespace {
 using namespace dnnl::impl::data_type;
 using namespace dnnl::impl::prop_kind;
 
-// clang-format off
-const std::map<pk_impl_key_t, std::vector<impl_list_item_t>> impl_list_map REG_BNORM_P({
-    {{forward}, {
-        /* fp */
-        CPU_INSTANCE_X64(jit_uni_batch_normalization_fwd_t<avx512_common>)
-        CPU_INSTANCE_X64(jit_uni_batch_normalization_fwd_t<avx2>)
-        CPU_INSTANCE_X64(jit_uni_batch_normalization_fwd_t<sse41>)
-        CPU_INSTANCE_X64(jit_uni_tbb_batch_normalization_fwd_t<avx512_common>)
-        CPU_INSTANCE_X64(jit_uni_tbb_batch_normalization_fwd_t<avx2>)
-        CPU_INSTANCE_X64(jit_uni_tbb_batch_normalization_fwd_t<sse41>)
-        CPU_INSTANCE_AARCH64(jit_uni_batch_normalization_fwd_t<sve_512>)
-        CPU_INSTANCE_AARCH64(jit_uni_batch_normalization_fwd_t<asimd>)
-        CPU_INSTANCE(ncsp_batch_normalization_fwd_t<f32>)
-        CPU_INSTANCE(ncsp_batch_normalization_fwd_t<bf16>)
-        CPU_INSTANCE(nspc_batch_normalization_fwd_t<f32>)
-        CPU_INSTANCE(nspc_batch_normalization_fwd_t<bf16>)
-        CPU_INSTANCE(ref_batch_normalization_fwd_t<f32>)
-        CPU_INSTANCE(ref_batch_normalization_fwd_t<bf16>)
-        /* int */
-        CPU_INSTANCE_X64(jit_uni_batch_normalization_s8_fwd_t<avx512_core>)
-        CPU_INSTANCE_X64(jit_uni_batch_normalization_s8_fwd_t<avx2>)
-        CPU_INSTANCE_X64(jit_uni_batch_normalization_s8_fwd_t<sse41>)
-        CPU_INSTANCE_AARCH64(jit_uni_batch_normalization_s8_fwd_t<sve_512>)
-        CPU_INSTANCE(ref_batch_normalization_fwd_t<s8>)
-        nullptr,
-    }},
-    {{backward}, REG_BWD_PK({
-        CPU_INSTANCE_X64(jit_uni_batch_normalization_bwd_t<avx512_common>)
-        CPU_INSTANCE_X64(jit_uni_batch_normalization_bwd_t<avx2>)
-        CPU_INSTANCE_X64(jit_uni_batch_normalization_bwd_t<sse41>)
-        CPU_INSTANCE_X64(jit_uni_tbb_batch_normalization_bwd_t<avx512_common>)
-        CPU_INSTANCE_X64(jit_uni_tbb_batch_normalization_bwd_t<avx2>)
-        CPU_INSTANCE_X64(jit_uni_tbb_batch_normalization_bwd_t<sse41>)
-        CPU_INSTANCE_AARCH64(jit_uni_batch_normalization_bwd_t<sve_512>)
-        CPU_INSTANCE_AARCH64(jit_uni_batch_normalization_bwd_t<asimd>)
-        CPU_INSTANCE(ncsp_batch_normalization_bwd_t<f32>)
-        CPU_INSTANCE(ncsp_batch_normalization_bwd_t<bf16>)
-        CPU_INSTANCE(nspc_batch_normalization_bwd_t<f32>)
-        CPU_INSTANCE(nspc_batch_normalization_bwd_t<bf16>)
-        CPU_INSTANCE(ref_batch_normalization_bwd_t<f32>)
-        CPU_INSTANCE(ref_batch_normalization_bwd_t<bf16>)
-        nullptr,
-    })},
-});
-// clang-format on
+const std::map<pk_impl_key_t, std::vector<impl_list_item_t>> &impl_list_map() {
+    // clang-format off
+    static std::map<pk_impl_key_t, std::vector<impl_list_item_t>> the_map = REG_BNORM_P({
```

Comment:
I can see that some maps are `static const` and others are just `static`. Maybe it would be nicer to have them all `static const`?

Reply:
They are dynamically initialised, so it's irrelevant. They're all in the heap and therefore read-write memory. So long as the function wrapping them retains the const it used to have, they can't be modified.

Reply:
+1 for this suggestion. It will be also nice to align `//clang-format off` and on - whether it's inside the method or outside, I see both versions implemented for different lists.
Rest LGTM, thanks for this improvement.

Reply:
It's more like a style thing, it would be nicer to keep it consistent.